src/api-server-selector.js

```diff
@@ -375,56 +373,53 @@ export class ApiServerSelector extends EventsTargetMixin(AmfHelperMixin(LitEleme
   renderExtraSlot() {
     const { extraOptions } = this;
     if (extraOptions) {
-      return html`
-        <slot name="api-server-extra-slot"></slot>
-      `;
+      return html`<slot name="api-server-extra-slot"></slot>`;
     }
     return undefined;
   }
 
-  _renderUrl() {
-    if (this._selectedValue === null || this._selectedValue === undefined) {
-      return;
-    }
-    const isCustom = this._getSelectedType(this._selectedIndex) === 'custom';
-
-    if (isCustom) {
-      return html`
-      <anypoint-input class="url-input" @input=${this._handleUrlChange} value="${this.url}">
-        <label slot="label">URL</label>
-      </anypoint-input>
-    `;
-    }
-    return html`
-    <anypoint-input class="url-input" disabled value="${this.url}">
-      <label slot="label">URL</label>
-      </anypoint-input>
-    `;
+  _handleUriChange(event) {
+    const { value } = event.target;
+    this.uri = value;
   }
 
-  _handleUrlChange(event) {
-    const { value } = event.target;
-    this.url = value;
+  _resetSelection() {
+    this._changeSelected({ selectedIndex: undefined, selectedValue: undefined });
   }
 
   render() {
     const { _selectedIndex, inline } = this;
+    const isCustom = this._selectedType === 'custom';
     return html`<style>${this.styles}</style>
     <div class="${inline ? 'inline' : ''}">
-      <anypoint-dropdown-menu class="api-server-dropdown">
-        <label slot="label">Select server</label>
-        <anypoint-listbox
-          .selected="${_selectedIndex}"
-          @selected-changed="${this.handleSelectionChanged}"
-          slot="dropdown-content"
-          tabindex="-1"
+      ${isCustom
+        ? html`<anypoint-input class="uri-input" @input=${this._handleUriChange} value="${this.uri}">
```

Comment:
Can you split it into separate template methods? It's for readability and possible future extensibility if you use this as a base class of another element.